torch/csrc/api/include/torch/nn/functional/vision.h

```diff
@@ -46,6 +47,21 @@ inline Tensor affine_grid(
   return torch::affine_grid_generator(theta, size, align_corners);
 }
 
+inline Tensor grid_sample(
+    const Tensor& input,
+    const Tensor& grid,
+    GridSampleOptions options = {}) {
+  
+  if (!options.align_corners().has_value()) {
+    TORCH_WARN("Default grid_sample and affine_grid behavior will be changed ",
+                "to align_corners=False from 1.4.0. ",
+                "See the documentation of grid_sample for details.");
+    options.align_corners(true);
+  }
+  
+  return torch::grid_sampler(input, grid, static_cast<int64_t>(options.mode()), static_cast<int64_t>(options.padding_mode()), options.align_corners().value());
```

Comment:
It would be awesome to directly translate the Python logic to the C++ version:
https://github.com/pytorch/pytorch/blob/783c9c84450f9b75d8cb9752062e961b7449e2c6/torch/nn/functional.py#L2675-L2702
including the temporary value `mode_enum` and `padding_mode_enum` for the enums. Thanks so much for your help! :D